kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/inband/KeyContext.java

```diff
@@ -67,14 +74,19 @@ public int encodedSize(int plaintextSize) {
      * @param output The output buffer
      */
     public void encode(@NonNull ByteBuffer plaintext, @NonNull ByteBuffer output) {
-        if (remainingEncryptions.get() < 0) {
+        if (remainingEncryptions <= 0) {
             throw new ExhaustedDekException("No more encryptions");
         }
         encryptor.encrypt(plaintext, output);
     }
 
     @Override
     public void destroy() throws DestroyFailedException {
```

Comment:
It's weird that we only override on method of the two present in the Destroyable interface.

Reply:
Implemented the other half and used it to replace isAlive